admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/helper/ClientSettingsHelper.java

```diff
@@ -151,13 +151,18 @@ public Map<Class, CompletableFuture> getInitiateDataFetch(String machineId, Stri
 		// invokes master-data-service
 		futuresMap.put(LocationHierarchy.class,
 				hasURLDetails(LocationHierarchy.class, isV2API, deltaSync) ? getURLDetails(LocationHierarchy.class)
-						: serviceHelper.getLocationHierarchyList(lastUpdated));
+						: serviceHelper.getLocationHierarchyList(getLastUpdatedTimeFromEntity(LocationHierarchy.class, lastUpdated, fullSyncEntities)));
 		futuresMap.put(DynamicFieldDto.class,
 				hasURLDetails(DynamicFieldDto.class, isV2API, deltaSync) ? getURLDetails(DynamicFieldDto.class)
-						: serviceHelper.getAllDynamicFields(lastUpdated));
+						: serviceHelper.getAllDynamicFields(getLastUpdatedTimeFromEntity(DynamicFieldDto.class, lastUpdated, fullSyncEntities)));
 
 		return futuresMap;
 	}
+	
+	private LocalDateTime getLastUpdatedTimeFromEntity(Class clazz, LocalDateTime lastUpdated, String fullSyncEntities) {
+		List<String> entities = Arrays.asList(fullSyncEntities.split("\\s*,\\s*"));
```

Comment:
This list is created for each entity, instead lets split and create the list only once. Pass the list this method.

Reply:
Modified code as suggested.